### PR TITLE
UI tests

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Add UI tests and pipeline

**Related issue(s)**

### CODE REVIEW

File: tests/ui/tests/scripts/kyma-dashboard-integration.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function build_and_run_busola() {
+echo "Create k3d registry..."
+k3d registry create registry.localhost --port=5000
+
+echo "Running kyma-dashboard..."
+docker run -d --rm --net=host --pid=host --name kyma-dashboard "$DASHBOARD_IMAGE"
+
+echo "waiting for server to be up..."
```
Comment (nataliasitko) [inline tests/ui/tests/scripts/kyma-dashboard-integration.sh:64]:
```suggestion
echo "Waiting for the server to be up..."
```

File: tests/ui/tests/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+## Installation
+
+To install dependencies, run the `npm install` command.
+
+## Development
+
+### Run Cypress UI tests in the headless mode
+
+To run Cypress UI tests using a Chrome browser in `headless mode`,
+pointing to a `remote Kyma-Dashboard` cluster with the default `local.kyma.dev` domain, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:26]:
```suggestion
pointing to a remote Kyma Dashboard cluster with the default `local.kyma.dev` domain, use this command:
```

File: tests/ui/tests/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+npm test
+```
+
+To run the tests, pointing to a `remote Kyma-Dashboard` cluster with `custom` domain, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:32]:
```suggestion
To run the tests, pointing to a remote Kyma Dashboard cluster with a custom domain, use this command:
```

File: tests/ui/tests/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+To run the tests, pointing to a `local Kyma-Dashboard` instance, use this command:
+
+```bash
+npm run test:local
+```
+
+### Run Cypress UI tests in the test runner mode
+
+To open Cypress UI `tests runner`,
+pointing to a `remote Kyma-Dashboard` cluster with the default `local.kyma.dev` domain, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:47]:
```suggestion
pointing to a remote Kyma Dashboard cluster with the default `local.kyma.dev` domain, use this command:
```

File: tests/ui/tests/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+npm test
+```
+
+To run the tests, pointing to a `remote Kyma-Dashboard` cluster with `custom` domain, use this command:
+
+```bash
+CYPRESS_DOMAIN={YOUR_DOMAIN} npm test
+```
+
+To run the tests, pointing to a `local Kyma-Dashboard` instance, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:38]:
```suggestion
To run the tests, pointing to a local Kyma Dashboard instance, use this command:
```

File: tests/ui/tests/README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+### Run Cypress UI tests in the test runner mode
+
+To open Cypress UI `tests runner`,
+pointing to a `remote Kyma-Dashboard` cluster with the default `local.kyma.dev` domain, use this command:
+
+```bash
+npm run start
+```
+
+To open the `tests runner`, pointing to a `remote Kyma-Dashboard` cluster with `custom` domain, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:53]:
```suggestion
To open the `tests runner`, pointing to a remote Kyma Dashboard cluster with a custom domain, use this command:
```

File: tests/ui/tests/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+CYPRESS_DOMAIN={YOUR_DOMAIN} npm run start
+```
+
+To open the `tests runner`, pointing to a `local Kyma-Dashboard` instance, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:59]:
```suggestion
To open the `tests runner`, pointing to a local Kyma Dashboard instance, use this command:
```

File: tests/ui/tests/README.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+npm run start:local
+```
+
+### Smoke tests
+
+To run smoke tests, pointing to a local Kyma-Dashboard instance, use this command:
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:67]:
```suggestion
To run smoke tests, pointing to a local Kyma Dashboard instance, use this command:
```

File: tests/ui/tests/scripts/kyma-dashboard-integration.sh [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+function deploy_k3d_kyma (){
+curl -Lo kyma https://storage.googleapis.com/kyma-cli-unstable/kyma-linux
+chmod +x ./kyma
+
+echo "Provisioning k3d cluster for Kyma"
+sudo ./kyma provision k3d --ci
+
+export KUBECONFIG=$(k3d kubeconfig merge kyma)
+./kyma deploy
```
[the remark this answers is not in the document]
Comment (barchw) [reply tests/ui/tests/scripts/kyma-dashboard-integration.sh:20]: Discussion: Install istio module

File: tests/ui/tests/support/login-commands.js [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    storage: null,
+    staticToken: false,
+  };
+  const { fileName, expectedLocation, storage, staticToken } = {
+    ...defaults,
+    ...params,
+  };
+
+  cy.wrap(loadFile('kubeconfig.yaml')).then(kubeconfig => {
+    if (kubeconfig.users?.[0]?.user?.exec?.args) {
```
[the remark this answers is not in the document]
Comment (triffer) [reply tests/ui/tests/support/login-commands.js:29]: We'll keep it, to support testing on Gardener.

File: tests/ui/tests/README.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+```bash
+npm run start
+```
+
+### Using a remote Kyma Dashboard instance
+
+#### Optional: Log in to a cluster using OIDC
+
+If a cluster requires OIDC authentication, include additional arguments `CYPRESS_OIDC_PASS` and `CYPRESS_OIDC_USER` while running the npm scripts.
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:47]:
```suggestion
If a cluster requires OIDC authentication, include the additional arguments **CYPRESS_OIDC_PASS** and **CYPRESS_OIDC_USER** while running the npm scripts.
```

File: tests/ui/tests/README.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
+# Tests
+
+## Overview
+
+This project contains API-Gateway integration tests for Kyma Dashboard.
```
Comment (nataliasitko) [inline tests/ui/tests/README.md:5]:
```suggestion
This project contains API Gateway integration tests for Kyma Dashboard.
```

**Review by kolodziejczak — APPROVED** [review]

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**